src/status_im/models/mailserver.cljs

```diff
@@ -0,0 +1,81 @@
+(ns status-im.models.mailserver
+  (:require
+   [clojure.string :as string]
+   [status-im.utils.handlers-macro :as handlers-macro]
+   [status-im.utils.ethereum.core :as ethereum]
+   [status-im.utils.inbox :as utils.inbox]
+   [status-im.data-store.mailservers :as data-store.mailservers]))
+
+(defn- extract-address-components [address]
+  (rest (re-matches #"enode://(.*)@(.*)" address)))
+
+(defn- get-chain [db]
+  (let [network  (get (:networks (:account/account db)) (:network db))]
+    (ethereum/network->chain-keyword network)))
+
+(defn- build [id mailserver-name address]
+  (assoc (utils.inbox/address->mailserver address)
+         :id (string/replace id "-" "")
+         :name mailserver-name))
+
+(defn build-url [address password]
+  (let [[initial host] (extract-address-components address)]
+    (str "enode://" initial ":" password "@" host)))
+
+(defn set-input [input-key value {:keys [db]}]
+  {:db (update db :mailservers/manage assoc input-key {:value value
+                                                       :error (case input-key
+                                                                :id   false
+                                                                :name (string/blank? value)
+                                                                :url  (not (utils.inbox/valid-enode-address? value)))})})
```

Comment:
80 chars

Reply:
oh i see that's not a new code...

Reply:
that's cool,I will fix that anyway, I introduced it probably before myself.